app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -73,6 +79,30 @@ fun <T> LiveData<T?>.observeNotNull(owner: LifecycleOwner, observer: (t: T) -> U
     observe(owner) { it?.let(observer) }
 }
 
+fun Recipient.displayedName(context: Context): String {
+    return if (AccountUtils.currentUser?.email == email) context.getString(R.string.contactMe) else getNameOrEmail()
+}
+
+fun Recipient.getNameOrEmail() = name?.ifBlank { email } ?: email
+
+fun <T> LiveData<T>.observeOnce(lifecycleOwner: LifecycleOwner, observer: Observer<T>) {
+    observe(lifecycleOwner, object : Observer<T> {
+        override fun onChanged(t: T?) {
+            observer.onChanged(t)
+            removeObserver(this)
+        }
+    })
+}
+
 fun Fragment.notYetImplemented() {
     showSnackbar("This feature is currently under development.")
 }
+
+fun Fragment.openMessageEdition(@IdRes direction: Int, message: Message? = null) {
+    val args = Bundle().apply {
+        putCharSequence("draftUuid", message?.draftUuid)
+        putCharSequence("draftResource", message?.draftResource)
+        putCharSequence("messageUid", message?.uid)
+    }
+    safeNavigate(direction, args)
+}
```

Comment:
We can do it differently with the navigation, using the class args generated.

Reply:
done